spec/controllers/sign_in/application_controller_spec.rb

```diff
@@ -63,6 +63,31 @@ def append_info_to_payload(payload)
       end
     end
 
+    shared_context 'user fingerprint validation' do
+      context 'user.fingerprint matches request IP' do
+        it 'passes fingerprint validation and does not create a log' do
+          expect_any_instance_of(SentryLogging).not_to receive(:log_message_to_sentry).with(:warn)
+          expect(subject.request.ip).to eq(user.fingerprint)
+        end
+      end
+
+      context 'user.fingerprint does not match request IP' do
+        let!(:user) { create(:user, :loa3, uuid: access_token_object.user_uuid) }
+        let(:expected_error) { SignIn::Errors::FingerprintMismatchError }
+        let(:log_context) { { request_ip: request.env['REMOTE_ADDR'], fingerprint: user.fingerprint } }
```

Comment:
Any reason not to refer to the ip as `request.ip` here? (instead of `request.env['REMOTE_ADDR']`)

Reply:
(And all of the other places `request.env['REMOTE_ADDR']` is used?)